### ISSUE TITLE
.argmax(axis = -1) not working propperly

### ISSUE THREAD

**Issue by weiserhase** [keras-team/keras#14021]
**System information**  
- Have I written custom code (as opposed to using example directory):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  2.2.0
- Keras version:  2.3.1
- Python version:  3.8.2
- CUDA/cuDNN version:  10.1.105
- GPU model and memory: Geforce Rtx 2060 Super 8Gb gddr5

**Describe the current behavior**  
im actualy trying to build a working face recognition using keras i got the moddel training and learing so i tried to integrate the prediction feature so i started with building the generator everythging worked well but when i wanted to extract the classes from the result i start to get Problems with this code:

`
test_generator = test_datagen.flow_from_directory(
    directory=test_dir,
    target_size=(img_width, img_height),
    color_mode="rgb",
    batch_size=1,
    class_mode=None,
    shuffle=False
)

test_generator.reset()
pred = (model.predict_generator(test_generator, verbose=1,steps=len(test_generator)))
print(pred)
predicted_class_indices = np.argmax(pred, axis=-1)
labels = (train_generator.class_indices)
print(predicted_class_indices)
labels = dict((v,k) for k,v in labels.items())
print(labels)
predictions = [labels[k] for k in predicted_class_indices]
filenames=test_generator.filenames
result = [predictions,filenames]
print(convert_to_class(pred))
print(result)
`

and thats what i get out

>  3/3 [==============================] - 0s 44ms/step

>[[0.933246  ]
 >[0.6150473 ]
 >[0.95384455]]
>[0 0 0]
>{0: 'angela merkel', 1: 'donald trump', 2: 'xxxxx'}
>[['angela merkel', 'angela merkel', 'angela merkel'], ['1\\000_1qk858.jpg', '1\\1-format430.jpg', '1\\IMG_20200501_045550.jpg']]

Like you see the tensor is [0,0,0] so all pictures are the same class 
Im now wondering what i messed up, even with some other ways suggested by people on stackoverflow etc. I got wrong results

**Comment by weiserhase** [comment]
I figured ot out i put the wrong class mode so it was set to binary and just returnt a binaty output 
`train_generator = train_datagen.flow_from_directory(
    train_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode='

> categorical

')

validation_generator = test_datagen.flow_from_directory(
    validation_data_dir,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode='categorical')
`
